api/common/request_id_util.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import (
+	"context"
+
+	"github.com/fnproject/fn/api/id"
+)
+
+const (
+	maxLength = 32
+)
+
+// FnRequestID returns the passed value if that is not empty or too long otherwise it generates a new unique ID
+func FnRequestID(ridFound string) string {
+	if ridFound == "" {
+		return id.New().String()
+	}
+	if len(ridFound) > maxLength {
+		// we truncate the rid to the value specified by maxLenght const
+		return ridFound[:maxLength]
+	}
+	return ridFound
+}
+
+//IncomingRID extract the request id from ctx
+func IncomingRID(ctx context.Context) string {
```

Comment:
`RequestIDFromContext` is more conventional, the preferred way to structure seems to be more something akin to `requestid.FromContext` where we'd have a `requestid` package, but it's unclear if we need/want that.

Reply:
Renamed to RequestIDFromContext, thanks